### PR TITLE
feat(auth-admin): View delegation and refactor access card

### PR DESCRIPTION
**Pull request by magnearun**
## What

Refactor AccessCard, now it displays actions depending on what functions are provided, if onDelete is provided a delete button is displayed etc, moving the logic about when different actions are displayed out side of the card.

General mandate delegations should only have the View action in "From me" list í Access control

Adds createdBy field to Delegations, used to display who created the general mandate delegations

Updates the View modal to dislpay referenceId and createdby when viewing a general mandate delegation. 

## Why

Now making changes to different actions on the card you can be confident you are not messing up what actions show up on other lists elsewhere. 

Match design and show correct actions in correct places

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced a loading state in the Create Delegation Confirm Modal for improved user feedback during operations.
  - Added a Delegation View Modal for displaying detailed information about delegations.
  - Implemented a Delegation Delete Modal for confirming deletion actions.

- **Improvements**
  - Updated delegation type labels for better clarity and consistency in the Create Delegation Screen.
  - Enhanced type safety for delegation types, ensuring more reliable operations within the delegation process.
  - Improved loading state management in the delegation creation flow for a smoother user experience.

- **Localization**
  - Continued enhancements to localization for better accessibility and user understanding, including new message definitions for delegation-related information.
  - Added new messages to support additional context about delegation creators and other relevant details.

### CODE REVIEW

File: libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx [hunk 1]
```diff
@@ -266,57 +254,61 @@ export const AccessCard = ({
...
-              {canDelete && (
+              {onDelete && (
                 <Button
                   variant="text"
                   icon="trash"
                   iconType="outline"
                   size="small"
                   colorScheme="destructive"
-                  onClick={() => onDelete(delegation)}
+                  onClick={() => onDelete?.(delegation)}
```
Comment (saevarma) [inline libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx:272]: Should be unnecessary to optional chain the function call as you have checked that `onDelete` is defined. 
```suggestion
                  onClick={() => onDelete(delegation)}
```

File: libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx [hunk 2]
```diff
@@ -266,57 +254,61 @@ export const AccessCard = ({
...
-                ) : null}
+                ) : (
+                  isExpired &&
+                  onRenew && (
+                    <Button
+                      icon="reload"
+                      iconType="outline"
+                      size="small"
+                      variant="utility"
+                      onClick={() => onRenew?.(delegation)}
```
Comment (saevarma) [inline libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx:306]: Unnecessary optional functional call chaining
```suggestion
                      onClick={() => onRenew(delegation)}
```

File: libs/portals/admin/delegation-admin/src/components/DelegationList.tsx [hunk 3]
```diff
@@ -3,43 +3,67 @@ import { Box, Stack } from '@island.is/island-ui/core'
...
-                  },
-                })
-                if (data) {
-                  revalidate()
+    <>
+      <Box marginTop={2}>
+        <Stack space={3}>
+          {delegationsList.map((delegation) => {
+            return (
+              <AccessCard
```
Comment (saevarma) [inline libs/portals/admin/delegation-admin/src/components/DelegationList.tsx:39]: We should reuse the `DelegationViewModal` in the admin portal to view the details of the delegation in the `onView` action of the card.

File: libs/portals/shared-modules/delegations/src/components/delegations/incoming/DelegationIncoming.graphql [hunk 4]
```diff
@@ -22,10 +22,19 @@ query AuthDelegationsIncoming($input: AuthDelegationsInput!, $lang: String) {
...
+      name
+    }
     from {
       nationalId
       name
     }
+    createdBy {
+      nationalId
+      name
+    }
```
Comment (saevarma) [inline libs/portals/shared-modules/delegations/src/components/delegations/incoming/DelegationIncoming.graphql:37]: Remove `referenceId` and `createdBy` here?

File: libs/portals/shared-modules/delegations/src/components/delegations/outgoing/DelegationsOutgoing.graphql [hunk 5]
```diff
@@ -24,10 +24,20 @@ query AuthDelegationsOutgoing($input: AuthDelegationsInput!, $lang: String) {
...
+    referenceId
     to {
       nationalId
       name
     }
+    from {
+      nationalId
+      name
+    }
+    createdBy {
```
Comment (saevarma) [inline libs/portals/shared-modules/delegations/src/components/delegations/outgoing/DelegationsOutgoing.graphql:37]: Remove `referenceId` and `createdBy`?

**Review by saevarma — APPROVED** [review]
🚀

File: libs/portals/shared-modules/delegations/src/components/delegations/outgoing/DelegationsOutgoing.graphql (hunk 5 above)
Comment (magnearun) [reply libs/portals/shared-modules/delegations/src/components/delegations/outgoing/DelegationsOutgoing.graphql:37]: yes

**Pull request merged**